BardMusicPlayer.Catalog/CatalogManager.cs

```diff
@@ -0,0 +1,359 @@
+﻿using BardMusicPlayer.Notate.Objects;
+using LiteDB;
+using System;
+using System.Collections.Generic;
+
+namespace BardMusicPlayer.Catalog
+{
+    public sealed class CatalogManager : IDisposable
+    {
+        private readonly LiteDatabase dbi;
+        private bool disposedValue;
+
+        /// <summary>
+        /// Internal constructor; this object is constructed with a factory pattern.
+        /// </summary>
+        /// <param name="dbi"></param>
+        private CatalogManager(LiteDatabase dbi)
+        {
+            this.dbi = dbi;
+            this.disposedValue = false;
+        }
+
+        /// <summary>
+        /// Create a new instance of the CatalogManager based on the given LiteDB database.
+        /// </summary>
+        /// <param name="dbPath"></param>
+        /// <returns></returns>
+        public static CatalogManager CreateInstance(string dbPath)
+        {
+            var dbi = new LiteDatabase(dbPath);
+            MigrateDatabase(dbi);
+
+            return new CatalogManager(dbi);
+        }
+
+        /// <summary>
+        /// This creates a playlist containing songs that match the given tag.
+        /// </summary>
+        /// <param name="tag"></param>
+        /// <returns></returns>
+        public IPlaylist CreatePlaylistFromTag(string tag)
+        {
+            if (tag == null)
+            {
+                throw new ArgumentNullException();
+            }
+
+            var songCol = this.GetSongCollection();
+
+            // TODO: This is brute force and not memory efficient; there has to be a better
+            // way to do this, but my knowledge of LINQ and BsonExpressions isn't there yet.
+            var allSongs = songCol.FindAll();
+            var songList = new List<MMSong>();
+
+            foreach (var entry in allSongs)
+            {
+                if (TagMatches(tag, entry))
+                {
+                    songList.Add(entry);
+                }
+            }
+
+            if (songList.Count == 0)
+            {
+                return null;
+            }
+
+            var dbList = new DBPlaylist()
+            {
+                Name = tag,
+                Songs = songList,
+                Id = null
+            };
+
+            return new DBPlaylistDecorator(dbList);
+        }
+
+        /// <summary>
+        /// This creates a new empty playlist with the given name.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <returns></returns>
+        public IPlaylist CreatePlaylist(string name)
```

Comment:
so what if a user calls this with multiple song names, but different files? they send it to the DB. then, update. instrument change, note change, try to send it again? it would be rejected.

Reply:
we need a playlist object. even for create